src/include/denc.h

```diff
@@ -763,6 +763,12 @@ struct denc_traits<bufferptr> {
     p.append(v);
   }
   template <class It>
+  static std::enable_if_t<!is_const_iterator_v<It>>
+  encode(bufferptr&& v, It& p, uint64_t f=0) {
+    denc((uint32_t)v.length(), p);
+    p.append(std::move(v));
+  }
+  template <class It>
```

Comment:
probably we could unify this variant with the `bufferptr&` one by templating them and perfect forwarding.  maybe this approach could also pave the road to the dencoder for `seastar::temporary_buffer`.